pkg/modules/api/context.go

```diff
@@ -126,6 +138,144 @@ func newContext(echoCtx echo.Context, logger *zap.Logger, fs *gotenberg.FileSyst
+		err = json.Unmarshal([]byte(raw[0]), &dls)
+		if err != nil {
+			return nil, cancel, WrapError(
+				fmt.Errorf("unmarshal json: %w", err),
+				NewSentinelHttpError(http.StatusBadRequest, fmt.Sprintf("Invalid 'downloadFrom' form field value: %s", err)),
+			)
+		}
+
+		eg, _ := errgroup.WithContext(ctx)
+		for i, dl := range dls {
+			eg.Go(func() error {
+				deadline, ok := ctx.Deadline()
+				if !ok {
+					// Should not happen, as context is created with a timeout.
+					return errors.New("context has no deadline")
+				}
+
+				if strings.TrimSpace(dl.Url) == "" {
+					return WrapError(
+						errors.New("empty download from URL"),
+						NewSentinelHttpError(http.StatusBadRequest, fmt.Sprintf("Invalid 'downloadFrom' form field entry %d: URL must be set", i)),
+					)
+				}
+
+				err := gotenberg.FilterDeadline(downloadFromCfg.allowList, downloadFromCfg.denyList, dl.Url, deadline)
+				if err != nil {
+					return fmt.Errorf("filter URL: %w", err)
+				}
+
+				logger.Debug(fmt.Sprintf("download file from '%s'", dl.Url))
+
+				req, err := retryablehttp.NewRequest(http.MethodGet, dl.Url, nil)
+				if err != nil {
+					return fmt.Errorf("create request to '%s': %w", dl.Url, err)
+				}
+
+				req.Header.Set("User-Agent", "Gotenberg")
+				for key, value := range dl.ExtraHttpHeaders {
+					req.Header.Set(key, value)
+				}
+				req.Header.Set(traceHeader, trace)
+
+				client := &retryablehttp.Client{
+					HTTPClient: &http.Client{
+						Timeout: time.Until(deadline),
+					},
+					RetryMax:     downloadFromCfg.maxRetry,
+					RetryWaitMin: time.Duration(1) * time.Second,
+					RetryWaitMax: time.Until(deadline),
+					Logger:       gotenberg.NewLeveledLogger(logger),
+					CheckRetry:   retryablehttp.DefaultRetryPolicy,
+					Backoff:      retryablehttp.DefaultBackoff,
+				}
+
+				resp, err := client.Do(req)
+				if err != nil {
+					return WrapError(
+						fmt.Errorf("download file from to '%s': %w", dl.Url, err),
+						NewSentinelHttpError(http.StatusBadRequest, fmt.Sprintf("Unable to download file from '%s': %s", dl.Url, err)),
+					)
+				}
+				defer func() {
+					err := resp.Body.Close()
+					if err != nil {
+						logger.Error(fmt.Sprintf("close response body from '%s': %s", dl.Url, err))
+					}
+				}()
+
+				if resp.StatusCode != http.StatusOK {
+					return WrapError(
+						fmt.Errorf("download file from to '%s': got status: '%s'", dl.Url, resp.Status),
+						NewSentinelHttpError(http.StatusBadRequest, fmt.Sprintf("Unable to download file from '%s': got status: '%s'", dl.Url, resp.Status)),
+					)
+				}
+
+				contentDisposition := resp.Header.Get("Content-Disposition")
+				if contentDisposition == "" {
+					return WrapError(
+						fmt.Errorf("no 'Content-Disposition' header from '%s'", dl.Url),
+						NewSentinelHttpError(http.StatusBadRequest, fmt.Sprintf("No 'Content-Disposition' header from '%s'", dl.Url)),
+					)
+				}
+
+				_, params, err := mime.ParseMediaType(contentDisposition)
```

Comment:
@gulien this line is causing my application to fail with the error `mime: no media type`. My current file server returns the following:
```
Content-Disposition: ";filename*=UTF-8''tips-building-social-media-presence.pdf"
```

According to [rfc6266](https://www.rfc-editor.org/rfc/rfc6266#page-5), that's valid and the lack of disposition type should be treated as "attachment"

But the go ParseMediaType fails to parse that... any of these work instead:
```
Content-Disposition: "attachment;filename*=UTF-8''tips-building-social-media-presence.pdf"
Content-Disposition: "inline;filename*=UTF-8''tips-building-social-media-presence.pdf"
Content-Disposition: "#;filename*=UTF-8''tips-building-social-media-presence.pdf"
Content-Disposition: "anything_but_whitespace_here;filename*=UTF-8''tips-building-social-media-presence.pdf"
```

I will see if I can change my file server response headers... 

Do you have any experience contributing to `go`? I have never used it before. I'm wondering if instead of change my file server, I can [submit a patch here](https://github.com/golang/go/blob/165bf241f2f7c72cffd83e278d674ae3ddbd72a1/src/mime/mediatype.go#L105-L107)

Reply:
I logged https://github.com/golang/go/issues/69551 - if that seems reasonable to you and you're willing to give it a 👍 , I'd appreciate that

Reply:
I gave it a 👍 

Thanks for the investigation. I did not expect Golang to fail there 🫤 

In the meantime, it may be possible to force the "attachment"? For instance: https://sweemeng.medium.com/setting-content-disposition-in-s3-9598cefc2ea7

Reply:
Update - I confirmed that if I change my content-disposition header coming from the file server, this new code works for AWS S3 👍